chess/3-web-api/starter-code/server/Server.java

```diff
@@ -2,15 +2,12 @@
 
 import spark.*;
 
-import java.nio.file.Paths;
-
 public class Server {
 
     public int run(int desiredPort) {
         Spark.port(desiredPort);
 
-        var webDir = Paths.get(Server.class.getProtectionDomain().getCodeSource().getLocation().getPath(), "web");
-        Spark.externalStaticFileLocation(webDir.toString());
+        Spark.staticFiles.location("web");
```

Comment:
The problem with this is that it doesn't work for both Maven and running in IntelliJ. You basically reverted it back to what I had before I made the change to support the autograder.

Reply:
We need to figure out something that works on all the platforms.

Reply:
Are you sure this doesn't work in Maven? The autograder is in Maven and we haven't encountered any issues with this method.

Reply:
It has been a month now since I worked on it, but when I ran it in intelliJ it set the execution directory to a different place than when I ran it with Maven. So I moved to it to being relative to the location of the class file. If you have tested all of those permutations and you are sure it works, then let's change it back to the simplified code. Just let me know that you have verified it.

Reply:
@prattnj have you checked that your change works in all these environments:

1. Intellij execute run/debug for the client and server modules.
2. Maven running tests locally
3. Maven running tests in the autograder

If so then I would love to simplify this.

Reply:
It works for the autograder's own server, but I realize now that having multiple modules (the autograder itself doesn't) could be a variable that affects it. I try it but if it doesn't work, I believe I have another solution (using the `URI` class).

Reply:
I have tested and verified that `Spark.staticFiles.location("web");` works in the following situations:

- Running tests with Maven locally (Windows)
- Running tests in IntelliJ locally (Windows)
- Submitting a complete (or imcomplete) Phase 3 to the autograder, which uses Maven

Reply:
Thanks for doing all this extra work!

Reply:
Michael just told me about the case in which you said it wouldn't work, which is when the client is involved. I'm not sure why anything would change, since the web socket tests are still in the server module but I will test that as well.